core/src/test/java/io/questdb/test/griffin/engine/groupby/StableAwareStringHolderTest.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin.engine.groupby;
+
+import io.questdb.griffin.engine.groupby.GroupByAllocator;
+import io.questdb.griffin.engine.groupby.GroupByAllocatorArena;
+import io.questdb.griffin.engine.groupby.StableAwareStringHolder;
+import io.questdb.std.Chars;
+import io.questdb.std.Numbers;
+import io.questdb.std.str.DirectUtf16Sink;
+import io.questdb.std.str.StableDirectString;
+import io.questdb.test.AbstractCairoTest;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StableAwareStringHolderTest extends AbstractCairoTest {
+    @Test
+    public void testClearAndSet() throws Exception {
+        assertMemoryLeak(() -> {
+            try (GroupByAllocator allocator = new GroupByAllocatorArena(64, Numbers.SIZE_1GB)) {
+                StableAwareStringHolder holder = new StableAwareStringHolder();
+                holder.setAllocator(allocator);
+                holder.clearAndSet("foobar");
+                TestUtils.assertEquals("foobar", holder);
+
+                holder.clearAndSet("barbaz");
+                TestUtils.assertEquals("barbaz", holder);
+            }
+        });
+    }
+
+    @Test
+    public void testClearAndSetDirect() throws Exception {
```

Comment:
nit: would be great to add a test where `StableDirectSequence` is mixed with `String`. That's to make sure that the holder is future-proof and handles this correctly.

Reply:
good one